### DISCUSSION TITLE
Sonoff th16 with Moisture sensor MS01

### DISCUSSION THREAD

**Discussion by thetornado76** [arendst/Tasmota#14777] [Ideas]
Hello,
I am trying to get a Sonoff TH 16 to work with an MS01 moisture meter.
https://www.amazon.de/gp/product/B09KRQMCB9/ref=ppx_yo_dt_b_asin_title_o03_s00?ie=UTF8&psc=1
Tasmota v10.1.0 Noelle is installed on the Sonoff.
I am using SI7021 on GPIO 14 as a template. So far I have not received any values, do I need to configure anything else?
Many thanks for your help

**Comment by sfromis** [comment]
Someone with the sensor, and interest in getting it to work with Tasmota, will have to explore the sensor, and possibly create a driver to make it supported. That has not happened until now.

**Comment by thetornado76** [comment]
Thanks for the info. If I can assist anyone with driver development, please let me know.

**Reply by barbudor** [reply]
Sure,
open it up and tell us what's inside (good quality photos might also be needed)
However it may be something custom from Sonoff using a custom protocol and not standard chips. In that case some reverse engineering on the protocol would be needed (which would also need to operate the sensor with Sonoff original firmware on the TH16)

**Reply by sfromis** [reply]
I'd indeed expect the "usual" Sonoff TH setup with a small MCU in the sensor handling the details, and communicating that on an one-wire bus to the switch. Of course this is speculation.....

**Reply by barbudor** [reply]
I wonder why they bother using a custom 1-wire protocol for their T&H sensor while the jack has 2 GPIOs. So I2C directly hooked on a sensor would have been possible.
However, I don't think there is any soil moisture I2C sensor chip (with a public datasheet). Even the Adafruit one is using a Seesaw MCU (a SAMD9 Atmel ARM wiht an Adafruit firmware) to manage their capacitive soil sensor.

As the Adafruit Seesaw is supported, it could be an alternative option for your TH16 @thetornado76

**Reply by sfromis** [reply]
My guess for the "why" around that pesky one-wire connection is that they started with AM2301/DHT21, and while the TRRS plug allows for 2 data wires, only one is actually connected on the TH board. Easy to mod to connect the 2nd one, a bit strange that the original designer did not just do that. When they then decided to upgrade the probe to SI7021, they had to make it work with just one data wire.

**Reply by sfromis** [reply]
You may be interested in https://github.com/arendst/Tasmota/pull/15335

**Reply by florinssk** [reply]
any news ?

**Reply by ascillato** [reply]
> any news ?

Already implemented. See https://github.com/arendst/Tasmota/pull/15335#issuecomment-1094296528

**Comment by wolfn13** [comment]
@thetornado76 
Hi.
Did you get any informations to include the TH16 with MS01 (Moisture Sensor) via Tasmota to IO-Broker?

I would need the same...
